src/main/java/org/myrobotlab/codec/CodecUtils.java

```diff
@@ -270,20 +281,51 @@ public static String capitalize(final String line) {
      * using the selected JSON backend.
      *
      * @param json  The JSON to be deserialized in String form
-     * @param clazz The target class.
+     * @param clazz The target class. If a class is not supplied the default class returned will be an {@link #DEFAULT_OBJECT_TYPE}
      * @param <T>   The type of the target class.
      * @return An object of the specified class (or a subclass of) with the state
      * given by the json. Null is an allowed return object.
      * @throws JsonDeserializationException if an error during deserialization occurs.
      * @see #USING_GSON
      */
+    @SuppressWarnings("unchecked")
     public static <T> /*@Nullable*/ T fromJson(/*@Nonnull*/ String json, /*@Nonnull*/ Class<T> clazz) {
         try {
             if (USING_GSON) {
+                if (clazz == null) {
+                  clazz = (Class<T>)DEFAULT_OBJECT_TYPE;
+                }
                 return gson.fromJson(json, clazz);
+            } else {
+              if (clazz == null) {
+
+                JsonParser parser = mapper.getFactory().createParser(json);
+                
+                // "peek" at the next token to determine its type
+                JsonToken token = parser.nextToken();
+
+                if (token == JsonToken.START_OBJECT) {
+                  clazz = (Class<T>)Map.class;
+                } else if (token == JsonToken.START_ARRAY) {
+                  clazz = (Class<T>)ArrayList.class;
+                } else if (token.isScalarValue()) {
+                  JsonNode node = mapper.readTree(json);
+                  if (node.isInt()) {

```

Comment:
are we missing a bunch of types here?  boolean ? seems like we would want to flush out this method a bit.. no?

Reply:
good catch updated with a boolean type ..
also mapped isNumber a json type to Double Int still will be attempted first